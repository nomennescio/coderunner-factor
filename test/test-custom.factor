! Copyright 2019 nomennescio

USING: accessors debugger io kernel namespaces sequences tools.testest ;
IN: tests

! : passed. ( -- ) "Just passed" write ;
! : failed. ( -- ) "Just failed" write ;

: custom1 ( -- )

  [ "Just passed" write ] test-passed. set
  [ "Just failed" write drop ] test-failed. set

  "short strings" it#{
    <{ "Hello World!" -> "Hello World!" }>
    <{ "Hello Worlds!" -> "Hello Worlds!" }>
  }#
  
  "failing compares" it#{
    <{ "Hello World!" -> "Hello Worlds!" }>
    <{ "Hello Worlds!" -> "Hello Worlds! " }>
  }#
;

M: assert-sequence error.
  [ "Actually expected :" write expected>> seq. ]
  [ lf "but instead got :" write got>> seq. ] bi
;

: custom2 ( -- )

  [ "Test passed" write ] test-passed. set
  [ error. ] test-failed. set

  "short strings" it#{
    <{ "Hello World!" -> "Hello World!" }>
    <{ "Hello Worlds!" -> "Hello Worlds!" }>
  }#
  
  "failing compares" it#{
    <{ "Hello World!" -> "Hello Worlds!" }>
    <{ "Hello Worlds!" -> "Hello Worlds! " }>
  }#
;

: run-tests ( -- )
  "custom message" describe#{
    custom1
    custom2
  }#
;

MAIN: run-tests
