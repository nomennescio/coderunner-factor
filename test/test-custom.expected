
<DESCRIBE::>custom message

<IT::>short strings

<PASSED::>Just passed

<PASSED::>Just passed

<COMPLETEDIN::>0.129834 ms

<IT::>failing compares

<FAILED::>Just failed

<FAILED::>Just failed

<COMPLETEDIN::>0.076824 ms

<IT::>short strings

<PASSED::>Test passed

<PASSED::>Test passed

<COMPLETEDIN::>0.025016 ms

<IT::>failing compares

<FAILED::>Actually expected :<:LF:>"Hello Worlds!"<:LF:>but instead got :<:LF:>"Hello World!"

<FAILED::>Actually expected :<:LF:>"Hello Worlds! "<:LF:>but instead got :<:LF:>"Hello Worlds!"

<COMPLETEDIN::>0.162660 ms

<IT::>short strings

<PASSED::>Test Passed

<PASSED::>Test Passed

<COMPLETEDIN::>0.026153 ms

<IT::>failing compares

<FAILED::>Test Failed : Actually expected :<:LF:>"Hello Worlds!"<:LF:>but instead got :<:LF:>"Hello World!"

<FAILED::>Test Failed : Actually expected :<:LF:>"Hello Worlds! "<:LF:>but instead got :<:LF:>"Hello Worlds!"

<COMPLETEDIN::>0.055392 ms

<COMPLETEDIN::>0.726864 ms
